Networking stuff

Flaise opened the pull request:
- Added P2PSessionConnectFail callback.
- Removed the `Sync` requirement of the lobby list call result. It didn't appear to be necessary; the whole reason the callback manager is single threaded is to avoid putting this kind of restriction on external code, such as the Steam API itself.
- Added bindings to some ISteamGameServer functions: EnableHeartbeats, SetModDir, SetServerName, SetMapName, SetMaxPlayerCount.

Thinkofname commented on src/server.rs:
```diff
@@ -219,6 +220,49 @@ impl Server {
+        }
+    }
+
+	/// If your game is a "mod," pass the string that identifies it.  The default is an empty
```
Tab instead of spaces

Thinkofname commented on src/server.rs:
```diff
@@ -219,6 +220,49 @@ impl Server {
+        }
+    }
+
+	/// Set name of map to report in the server browser
```
Tab instead of spaces

Flaise commented:
Fixed the tabs. On my screen, Github says "This branch has no conflicts with the base branch". And I reran the appropriate git commands:
```
✔ ~/projects/steamworks-rs [p2p|✔] 
02:37 $ git checkout master
Switched to branch 'master'
Your branch is up to date with 'origin/master'.
✔ ~/projects/steamworks-rs [master|✔] 
08:15 $ git pull upstream master
From https://github.com/Thinkofname/steamworks-rs
 * branch            master     -> FETCH_HEAD
Already up to date.
✔ ~/projects/steamworks-rs [master|✔] 
08:15 $ git checkout p2p
Switched to branch 'p2p'
Your branch is up to date with 'origin/p2p'.
✔ ~/projects/steamworks-rs [p2p|✔] 
08:15 $ git merge master
Already up to date.
```
Are you sure there's something blocking a merge?

Thinkofname commented:
![image](https://user-images.githubusercontent.com/774399/63030027-887aa400-bea9-11e9-91b5-b621974c1390.png)

Seems to be because it defaults to rebase for me. The commits from your previous PRs are showing as new commits here because they were rebased when merged before unfortunately. I'll try and merge this one locally because its my fault for using rebase in the first place, i'll try not to do that anymore sorry!

Thinkofname commented:
Merged here: https://github.com/Thinkofname/steamworks-rs/compare/f109df710c37...a773be9f109b
